### PR TITLE
Add RELEASE_TEMPLATE.md

### PR DESCRIPTION
**Pull request by euri10**
prompted by https://github.com/urllib3/urllib3/issues/2307#issuecomment-868519361

I moved the existing PR template to the create dir too.

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -5,4 +5,4 @@ If this is your first PR to urllib3 please review the Contributing Guide:
 https://urllib3.readthedocs.io/en/latest/contributing.html
 
 Adhering to the Contributing Guide means we can review, merge, and release your change faster! :)
---->
+--->
```
Comment (sethmlarson) [inline .github/PULL_REQUEST_TEMPLATE/PULL_REQUEST_TEMPLATE.md:8]: Can we reduce this to a file move, looks like a line ending got tweaked here.

File: .github/PULL_REQUEST_TEMPLATE/RELEASE_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,4 @@
+> ### Before creating a release PR
```
Comment (sethmlarson) [inline .github/PULL_REQUEST_TEMPLATE/RELEASE_TEMPLATE.md:1]: We want the entire checklist, even all the stuff that doesn't have to do with creating the PR specifically.

Also I'm not sure we want to have the checklist quoted with `>`?

**Review by sethmlarson — CHANGES_REQUESTED** [review]
Thanks for this, two comments for you:

File: .github/PULL_REQUEST_TEMPLATE/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (euri10) [reply .github/PULL_REQUEST_TEMPLATE/PULL_REQUEST_TEMPLATE.md]: done !

File: .github/PULL_REQUEST_TEMPLATE/RELEASE_TEMPLATE.md (hunk 2 above)
Comment (euri10) [reply .github/PULL_REQUEST_TEMPLATE/RELEASE_TEMPLATE.md]: My bad, I quoted the original to copy and forgot to remove it :)

File: .github/PULL_REQUEST_TEMPLATE/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (sethmlarson) [reply .github/PULL_REQUEST_TEMPLATE/PULL_REQUEST_TEMPLATE.md]: Let's revert the change on this file.

File: .github/PULL_REQUEST_TEMPLATE/RELEASE_TEMPLATE.md (hunk 2 above)
Comment (sethmlarson) [reply .github/PULL_REQUEST_TEMPLATE/RELEASE_TEMPLATE.md]: Let's name this file `release.md`

**Comment by sethmlarson** [comment]
Could you capture the changes from the discussion on Discord:
- Remove the prose in our contributing.rst documentation in the releases section
- Instead replace it with short instructions including the URL: `https://github.com/urllib3/urllib3/compare/<BASE>...release-<VERSION>?expand=1&template=release.md` and a link to the release template under `.github/PULL_REQUEST_TEMPLATE/release.md`

**Comment by euri10** [comment]
> * Instead replace it with short instructions including the URL: `https://github.com/urllib3/urllib3/compare/<BASE>...release-<VERSION>?expand=1&template=release.md` and a link to the release template under `.github/PULL_REQUEST_TEMPLATE/release.md`

I'm not sure about the rst link, I removed the <> from your version, not sure if this can be escaped in rst

**Comment by pquentin** [comment]
I merged this pull request in my own fork and made a few tests. The problem with the checklist in the pull request is that you need a branch for the checklist to appear and to create the PR. See it for yourself:

 * This does not work: https://github.com/pquentin/urllib3/compare/main...main?expand=1&template=release.md.
 * But this works: https://github.com/pquentin/urllib3/compare/main...release-test?expand=1&template=release.md

This also means you can't have a link that just needs to be clicked, it needs to be manually crafted by adding `&template=release.md` to a PR that is already "expanded".

What can we do? I see those options:
 1. Use an issue for the checklist
 2. Move the first section of the checklist in `contributing.rst`

Option 2. here seems best: we're talking about maintainers here, so we should aim for efficiency rather than a low barrier to entry. And having the checklist in the release PR is efficient.

**Comment by euri10** [comment]
so if I follow correctly for option 2 I:
1. remove entirely  `.github/PULL_REQUEST_TEMPLATE/release.md` ?
2. put its contents into `contributing.rst`

**Comment by pquentin** [comment]
No, we keep `release.md` but:

 * move everything under `### Before creating a release PR` to `contributing.rst`
 * remove the `### After creating a release PR` line from `release.md`: we know we are in that pull request already anway.

**Review by sethmlarson — APPROVED** [review]
LGTM

**Pull request merged**